src/components/checkbox/checkbox.jsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import FormControl from '../form-control/form-control.jsx';
+import useMounted from '../../hooks/use-mounted.js';
+import Icon from '../icon/icon.jsx';
+import styles from './checkbox.css';
+import cautionSvg from '../../svgs/caution.svg';
+
+const Checkbox = forwardRef(function Checkbox(props, forwardedRef) {
+  const fallbackRef = useRef();
+  const ref = forwardedRef || fallbackRef;
```

Comment:
I copied this pattern from other components, but I think this can be shortened to just `const ref = useRef(forwardedRef && forwardedRef.current)`?

Reply:
React hooks cache their initialization. If we were to `useRef(forwardedRef && forwardedRef.current)` then it needs a `useEffect(() => ref.current = forwardedRef.current, [forwardedRef.current])`. But changes in refs don't propagate React renders... so there is a high likely hood that it will be stale.